vars/withPactTestOnlyPipeline.groovy

```diff
@@ -0,0 +1,145 @@
+import uk.gov.hmcts.contino.AngularPipelineType
+import uk.gov.hmcts.contino.azure.Acr
+import uk.gov.hmcts.contino.Builder
+import uk.gov.hmcts.contino.DockerImage
+import uk.gov.hmcts.contino.Environment
+import uk.gov.hmcts.contino.MetricsPublisher
+import uk.gov.hmcts.contino.NodePipelineType
+import uk.gov.hmcts.contino.PipelineType
+import uk.gov.hmcts.contino.ProjectBranch
+import uk.gov.hmcts.contino.SpringBootPipelineType
+import uk.gov.hmcts.contino.Subscription
+import uk.gov.hmcts.contino.AppPipelineConfig
+import uk.gov.hmcts.contino.AppPipelineDsl
+import uk.gov.hmcts.contino.PipelineCallbacksConfig
+import uk.gov.hmcts.contino.PipelineCallbacksRunner
+import uk.gov.hmcts.pipeline.AKSSubscriptions
+import uk.gov.hmcts.pipeline.deprecation.WarningCollector
+import uk.gov.hmcts.pipeline.TeamConfig
+
+def call(type, String product, String component, Closure body) {
+
+  def branch = new ProjectBranch(env.BRANCH_NAME)
+
+  def deploymentNamespace = branch.deploymentNamespace()
+  def deploymentProduct = deploymentNamespace ? "$deploymentNamespace-$product" : product
+
+  def pipelineTypes = [
+    java  : new SpringBootPipelineType(this, deploymentProduct, component),
+    nodejs: new NodePipelineType(this, deploymentProduct, component),
+    angular: new AngularPipelineType(this, deploymentProduct, component)
+  ]
+
+  Subscription subscription = new Subscription(env)
+  AKSSubscriptions aksSubscriptions = new AKSSubscriptions(this)
```

Comment:
Looks like, this has a few unused imports, variables. can you please review them ?